docs/azure/sdk/unit-testing-mocking.md

```diff
@@ -0,0 +1,348 @@
+---
+title: Unit testing and mocking with the Azure SDK .NET
+description: Learn techniques and tools for unit testing and mocking the Azure SDK for .NET
+ms.custom: devx-track-dotnet, engagement-fy23
+ms.date: 07/05/2023
+---
+
+# Unit testing and mocking with the Azure SDK .NET
+
+Unit testing is an important part of a sustainable development process that can improve code quality and prevent regressions or bugs in your app. However, unit testing presents challenges when the code you are testing communicates with an Azure service over a network. Tests that run against live services can experience issues such as latency that slows down test execution, dependencies on code outside of the isolated test, and issues with managing service state and costs every time the test is run. Instead of testing against live Azure services, replace the service clients with mocked or in-memory implementations. This avoids the above issues and lets developers focus on testing their application logic, independent from the network and service.
+
+In this article, we’ll show you how to use the Azure SDK to write great unit tests that isolate your dependencies to make your tests more reliable. First, we’ll show you how many of the Azure SDK building blocks have been designed to be replaced with an in-memory test implementation. Then, we’ll show you how the pieces come together into a fast and reliable unit test. Finally, we’ll provide some design tips to help you design your own classes to better support unit testing.
```

Comment:
Avoid "we", you're not doing this with the reader... 🙂 Also, seeing smart quotes, instead use normal quotes (was this copied from somewhere). You can turn on "Replace Smart Quotes" option that I wrote for the Docs Authoring Pack extension:

![image](https://github.com/dotnet/docs/assets/7679720/70649e5f-ff87-409d-a919-03f32ed81ba1)

```suggestion
In this article, you'll learn how to use the Azure SDK to write great unit tests that isolate your dependencies to make your tests more reliable. First, you'll see how many of the Azure SDK building blocks have been designed to be replaced with an in-memory test implementation. Then, you'll see you how the pieces come together into a fast and reliable unit test. Finally, you'll learn some design tips to help you design your own classes to better support unit testing.
```